externallib.php

```diff
@@ -126,9 +129,25 @@ public static function set_completion($jsondata) {
                 if ($record !== false && $record->get('enabled') == \local_annoto\completion::COMPLETION_TRACKING_AUTOMATIC) {
                     $status = true;
                     if ($completiondata = \local_annoto\completiondata::get_record(['completionid' => $record->get('id'), 'userid' => $USER->id])) {
-                        $completiondata->set('data', $jsondata);
-                        $completiondata->update();
-                        $message = "Update completion for user {$USER->id} modid {$data->cmid} completion {$data->completion}";
+                        $Act_comp = $DB->get_record('local_annoto_completion',array('cmid'=>$data->cmid));
+                        $current_time = time();
+                        // echo $Act_comp->completionexpected;
+                        // echo "------------";
+                        // echo time();
+                        // echo $Act_comp->view; && 
+                        // print_r($data);
+                        // die; 
+                        if(($data->completion >= (int)$Act_comp->view) && ($data->comments >= (int)$Act_comp->comments) && ($data->replies >= (int)$Act_comp->replies) && (($Act_comp->completionexpected != 0 && $current_time <= $Act_comp->completionexpected) ||  $Act_comp->completionexpected == 0)){
```

Comment:
code formatting not according to moodle standards.
Make more readable, line too long